src/rfc3986/validators.py

```diff
@@ -273,13 +285,15 @@ def is_valid(value, matcher, require):
         Whether or not the value is required.
     """
     if require:
-        return value is not None and matcher.match(value)
+        return value is not None and bool(matcher.match(value))
```

Comment:
These `bool(...)` additions are meant to serve the purpose of typing this function as returning `bool`, which is the intention, based on the docstring. Unless the returning of a `re.Match` object from here is an implementation detail that's relied upon, I hope this as harmless as it feels and within the scope of this PR to match the annotated return type with the actual return type.